pepicrft/catalysis-framework
# Standardize and document throwing and handling of errors across the codebase

## What?
I'm adding some foundational blocks for making errors a first-class piece in this codebase.

## Why?
**Errors can and will happen.** In many software projects they are often treated as an after-thought. However, that's a mistake because it might negatively impact the developer experience (DX) of the tool. Therefore we don't want to make that mistake in GestaltJS. Errors are first-class in this project. We'll have a well-defined set of errors with a conventional structure and a standard formatting.

## How?

- Implement a set of errors to abort the execution of the CLI.
- Implement formatting for those errors.
- Document best practices around errors and connect it to the errors existing in the codebase.

## Testing?

1. Make a command fail, for example adding this code to a command:

```js
throw new error.Bug('Message', {cause: "We couldn't generate whatever"})
```
2. Run the command.
3. It should format the error
